### PR TITLE
Fix red tree

### PR DESCRIPTION
**Pull request by dnfield**
Previous patch failed becuase it would allow completer to be completed multiple times - this fixes that.

### CODE REVIEW

File: dev/devicelab/bin/tasks/run_machine_concurrent_hot_reload.dart [hunk 1]
```diff
@@ -65,8 +65,8 @@ void main() {
         } else if (json != null && json['event'] == 'app.started') {
           appId = json['params']['appId'];
         }
-        if (vmServicePort != null && appId != null) {
-          print('run: ready!');
+        if (vmServicePort != null && appId != null && !ready.isCompleted) {
+          print('run: ready!');q
```
Comment (jonahwilliams) [inline dev/devicelab/bin/tasks/run_machine_concurrent_hot_reload.dart]: q?

Comment (dnfield) [reply dev/devicelab/bin/tasks/run_machine_concurrent_hot_reload.dart]: Hah. Thanks.

Comment (dnfield) [reply dev/devicelab/bin/tasks/run_machine_concurrent_hot_reload.dart]: I hit Ctrl+Z one too many times ;)

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Review by liyuqian — APPROVED** [review]

**Pull request merged**